examples/silicon_timer.jl

```diff
@@ -0,0 +1,39 @@
+using DFTK
```

Comment:
I wouldn't have this as a separate example. Either remove it or just add the timers to another example

Reply:
What do you mean ? To remove the `silicon_timer.jl` example and use timers in another example like `silicon.jl` ?

Reply:
Yeah. Or preferably even a more developer-oriented one like 1D GPE or https://github.com/JuliaMolSim/DFTK.jl/pull/246